02-peer-protocol.md

```diff
@@ -938,6 +960,38 @@ previously sent.
 A receiving node MAY ignore spurious message retransmission, or MAY
 fail the channel if they occur.
 
+### Rationale
+
+The effect of requirements above are that the opening phase is almost
+atomic: if it doesn't complete, it starts again.  The only exception
+is where the `funding_signed` message is sent and not received: in
+this case, the funder will forget the channel and presumably open
+a new one on reconnect; the other node will eventually forget the
+original channel due to never receiving `funding_locked`.
+
+For normal operation and during shutdown, the updates are similarly
+forgotten, though in this case the loss of `revoke_and_ack` can cause
+an entire retransmission of the previous updates and commitment:
+this is OK, as the recipient will simply ignore duplicates (as
+documented in the requirements for [`update_add_htlc`](#adding-an-htlc-update_add_htlc) and [Removing an HTLC](#removing-an-htlc-update_fulfill_htlc-update_fail_htlc-and-update_fail_malformed_htlc).
+
+This retransmission is required to generally be a super-set of the
+original, to give the same result whether the first transmission was
+committed or not.  The exception for `update_fee` applies because
+`update_fee` simply overrides previous `update_fee` messages, though a
+retransmission has to send at least one `update_fee` message if one was
+sent originally.
+
+Note that if no `commitment_signed` was sent in the first place,
+there's no requirement to retransmit those (uncommitted) updates.
```

Comment:
My point was that it would indeed allow the receiver to not discard changes, and just make sure that whatever it receives afterwards is either new or already known. It is more error prone than rollbacking IMO, and conceptually I find it easier to see a reconnection as a simple replay.